### ISSUE TITLE
Inconsistency in Reminder - not coming, sometime delayed

### ISSUE THREAD

**Issue by Jineshpanchal** [iSoron/uhabits#2052]
## Pre-submission checklist
- [yes ] I am submitting a bug report, not a feature request.
- [ yes] I have searched for similar issues, but did not find any matches.

## Description
We have been testing app in multiple scenario for reminder features as mentioned below and also on different android version. 

scenario
1. Screen in ON
2. Screen is ON but app is killed
3. Screen in OFF
4. Screen in OFF and APP is killed

Android version : 
Version 12 and version 14

In above 4 scenario, we have been testing the app but it is very random and not consistent. we have also tried to create build after forking the app and create our own APK to verify but behaviours is same and inconsistent. 

Could you please guide us, how to handle reminder consistency.

Also 2nd Observation is,
Sometime reminder trigger on exact time and sometime it triggers after few seconds (in our observation we hae seen the delay from 5 seconds to 50 second as well, but we dont know how to reproduce this scenario)

we really like this app and it is perfect, yet simple app for anyone to use. but only because of reminder is not coming on time and sometime delayed and in few scenario it is not even triggering. if we can address this issue, or if you can give us some time so we can explain our use case as well. 

We are from one spiritual NGO, and we wanted to add one feature in this app, where we want to invoke song when its time for reminder and that feature is also working but challenge is as mentioned above (it is not coming, sometime get delayed, and sometime working based on 4 scenarios mentioned above)

as a spiritual NGO, we are also contributing as much as we can to help community and your support will be hight appreciated.

## Steps to reproduce
1. Go to ...
6. Click on ....
7. Scroll down to ....
8. See error

## System information
 - Phone: [e.g. Google Pixel 4]
 - Phone Operating System: [e.g. Android 10]
 - App version: [e.g. 2.0.2]

**Comment by eren8204** [comment]
@Jineshpanchal @iSoron should i start working on it

**Comment by Jineshpanchal** [comment]
Thank you for offering yourself. Let us know if any help required.
you can also connect with @raaghavrm who is helping as well to get the feature done.

@raaghavrm Kindly share more details and steps to start.

**Comment by raaghavrm** [comment]
@eren8204 Hey, We can connect to solve the bug together.

**Comment by eren8204** [comment]
@raaghavrm sure

**Comment by iSoron** [comment]
@Jineshpanchal Thanks for using the app. Any pull requests improving the reliability of reminders is very welcome, but I am unfortunately not very positive it can be done. There's more information about this problem in the [don't kill my app!](https://dontkillmyapp.com/problem) website.

Regarding notification delays, Loop is already using exact alarms which, according to the Android documentation, should go off at a "precise moment in the future."

**Comment by eren8204** [comment]
@iSoron should we proceed or not?

**Comment by Jineshpanchal** [comment]
Yes brother @eren8204 , absolutely, if we succeed in fixing this challenge then we will get blessing of all the app users..

@iSoron has put lot of effort to make this app perfect in most of the usecase and you can help in fixing this one.

**Comment by eren8204** [comment]
@Jineshpanchal ok, i'll start and will let u know if i need some help

**Comment by Jineshpanchal** [comment]
@eren8204 did you get chance to work on and get workaround to fix it.

**Comment by eren8204** [comment]
@Jineshpanchal sorry, i was a little occupied, i'll update you soon

**Comment by eren8204** [comment]
@Jineshpanchal i tried to recreate the error in Pixel 8 Pro Android 14, but it was working perfectly in every scenario mentioned by you 
1. Screen in ON
2. Screen is ON but app is killed
3. Screen in OFF
4. Screen in OFF and APP is killed

**Comment by Jineshpanchal** [comment]
oh, may be because of pixel phone is by android/google only.
can we try to check in any other phone

**Comment by eren8204** [comment]
@Jineshpanchal i checked in personal redmi phone, it was working perfectly, can you help me replicate it otherwise how will i debug it

**Comment by raaghavrm** [comment]
@eren8204 We just installed in our phone(Vivo and Redmi, android version 12 and 14 respectively) and in both we faced issues.

**Comment by eren8204** [comment]
@iSoron was this actually fixed??
